dbt/include/trino/macros/materializations/incremental/incremental.sql

```diff
@@ -0,0 +1,69 @@
+{% macro dbt_trino_validate_get_incremental_strategy(config) %}
+  {#-- Find and validate the incremental strategy #}
+  {%- set strategy = config.get("incremental_strategy", default="append") -%}
+
+  {% set invalid_strategy_msg -%}
+    Invalid incremental strategy provided: {{ strategy }}
+    Expected one of: 'append', 'delete+insert'
+  {%- endset %}
+  {% if strategy not in ['append', 'delete+insert'] %}
+    {% do exceptions.raise_compiler_error(invalid_strategy_msg) %}
+  {% endif %}
+
+  {% do return(strategy) %}
+{% endmacro %}
+
+{% macro dbt_trino_get_incremental_sql(strategy, tmp_relation, target_relation, unique_key, dest_columns) %}
+  {% if strategy == 'append' %}
+    {% do return(get_append_sql(target_relation, tmp_relation, dest_columns)) %}
+  {% elif strategy == 'delete+insert' %}
+    {% do return(get_delete_insert_merge_sql(target_relation, tmp_relation, unique_key, dest_columns)) %}
+  {% elif strategy == 'merge' %}
+    {#-- Not yet supported by trino (blocked in dbt_trino_validate_get_incremental_strategy macro) --#}
+    {% do return(get_merge_sql(target_relation, tmp_relation, unique_key, dest_columns)) %}
+  {% endif %}
+{% endmacro %}
+
+{% macro get_append_sql(target_relation, tmp_relation, dest_columns) %}
```

Comment:
Is leaving `dbt_trino_` prefix done on purpose?

Reply:
I think they did that in the dbt-snowflake adapter to _prematurely_ avoid name clashes with internal dbt core macros. Remember all macros are loaded onto the same Jinja environment and name clashes are resolved based on load order.

Reply:
I'm advocating that we should have a consistent way of defining macros in the adapter. 
Any form that keeps the existing functionality is fine, but should be consistent across the codebase of the adapter. (obviously this should be a follow-up PR).

Reply:
Let's go for the removal of the prefix for now